build-support/bin/generate_github_workflows.py

```diff
@@ -326,6 +329,7 @@ def test_workflow_jobs(primary_python_version: str, *, cron: bool) -> Jobs:
             "name": "Bootstrap Pants, test Rust (macOS)",
             "runs-on": MACOS_VERSION,
             "strategy": {"matrix": {"python-version": [primary_python_version]}},
+            "timeout-minutes": 20,
```

Comment:
Sorry I was on my phone when I reviewed the first time and it was confusing which was which. This should be 40 minutes. Likewise, the Bootstrap Pants Linux shard should be 40 minutes and it looks like it doesn't have anything.